modules/core/src/main/scala/org/scalasteward/core/vcs/VCSExtraAlg.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2018-2019 Scala Steward contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.scalasteward.core.vcs
+
+import cats.effect.Async
+import cats.implicits._
+import org.scalasteward.core.data.Update
+import org.scalasteward.core.util.HttpExistenceClient
+import org.scalasteward.core.vcs
+
+trait VCSExtraAlg[F[_]] {
+  def getBranchCompareUrl(maybeRepoUrl: Option[String], update: Update): F[Option[String]]
+}
+
+object VCSExtraAlg {
+  def create[F[_]](
+      implicit
+      existenceClient: HttpExistenceClient[F],
+      F: Async[F]
```

Comment:
Why does this require `Async[F]`? It seems to me that `Monad[F]` would be enough.

Reply:
No intention, I just refered my memory from last PR 😄 
Changed to weaker type classes in https://github.com/fthomas/scala-steward/pull/814/commits/869d65beb23d86dc7024f41b9e1dcadac3698b15